src/Shared/Model/Metadata/NuGetProtocolMetadata.cs

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+
+namespace Microsoft.CST.OpenSource.Model.Metadata;
+
+using Contracts;
+using Newtonsoft.Json;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.Protocol;
+using NuGet.Versioning;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+/// <summary>
+/// A class to represent Package Metadata for a NuGet package.
+/// </summary>
+public record NuGetMetadata : IManagerMetadata
```

Comment:
Nit: should we emphasize the package-version-specific nature of this metadata? Maybe call it something like `NuGetPackageVersionMetadata`?

Reply:
I renamed it, and I ended up renaming the interface to `IManagerPackageVersionMetadata`